sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1644 @@
+(buff-to-int-be 0x00000000000000000000000000000001) ;; Returns 1
+(buff-to-int-be 0xffffffffffffffffffffffffffffffff) ;; Returns -1
+(buff-to-int-be 0x) ;; Returns 0
+```
+
+### New method: `buff-to-uint-be`
+
+* **Input Signature:** `(buff-to-uint-be (input (buff 16)))`
+* **Output Signature:** `uint`
+
+Converts a byte buffer to an unsigned integer use a big-endian
+encoding.  The byte buffer can be up to 16 bytes in length. If there
+are fewer than 16 bytes, as this function uses a big-endian encoding,
+the input behaves as if it is zero-padded on the _left_.
+
+**Rationale:** This method facilitates parsing user-supplied encoded data, or
+data available on-chain that is currently represented as a `buff`.
+While this functionality could have been implemented in Clarity, it would
+have been much more expensive to use.
+
+**Examples:**
+
+```clarity
+(buff-to-uint-be 0x01) ;; Returns u1
+(buff-to-uint-be 0x00000000000000000000000000000001) ;; Returns u1
+(buff-to-uint-be 0xffffffffffffffffffffffffffffffff) ;; Returns u340282366920938463463374607431768211455
+(buff-to-uint-be 0x) ;; Returns u0
+```
+
+### New method: `stx-transfer-memo?`
+
+* **Input Signature:** `(stx-transfer? (amount uint) (sender principal) (recipient principal) (memo buff))`
+* **Output Signature:** `(response bool uint)`
+
+`stx-transfer-memo?` is similar to `stx-transfer?`, except that it
+adds a `memo` field.
+
+This function returns `(ok true)` if the transfer
+is successful, or, on an error, returns the same codes as
+`stx-transfer?`.
+
+**Rationale:** This method facilitates integration with exchanges.  Exchanges
+often require a memo field for transferring funds into and out of their wallets,
+often for the purposes of multiplexing one address into many user accounts.
+Adding a native `stx-transfer-memo?` function makes it easy for smart contracts
+to interact with exchanges that follow this convention.
+
+**Examples:**
+
+```clarity
+(as-contract
+  (stx-transfer? u50 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY tx-sender 0x00)) ;; Returns (err u4)
+(stx-transfer-memo? u60 tx-sender 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY 0x010203)) ;; Returns (ok true)
+```
+
+### New method: `is-standard`
+
+* **Input Signature:** `(is-standard (standard-or-contract principal))`
+* **Output Signature:** `bool`
+
+Tests whether `standard-or-contract` _matches_ the current network
+type, and therefore represents a principal that can spend tokens on the current
+network type. That is, the network is either of type `mainnet`, or `testnet`.
+Only `SPxxxx` and `SMxxxx` _c32check form_ addresses can spend tokens on
+a mainnet, whereas only `STxxxx` and `SNxxxx` _c32check forms_ addresses can spend
+tokens on a testnet. All addresses can _receive_ tokens, but only principal
+_c32check form_ addresses that match the network type can _spend_ tokens on the
+network.  This method will return `true` if and only if the principal matches
+the network type, and false otherwise.
```

Comment:
This function should verify only if provided principal is standard or contract. Nothing else. One function - one responsibility.
Verification if principal can spend tokens in current network should be handled by separate function like `is-network-principal` proposed above.

Reply:
> This function should verify only if provided principal is standard or contract. Nothing else. One function - one responsibility.

Why?

Reply:
Because this is exactly what its name suggests. `is-standard` doesn't say anything about validation against network. 

Besides if I understand your https://github.com/stacksgov/sips/pull/95#discussion_r1007362805 correctly you can send funds to address that belongs to a different network resulting funds loss.
So instead of introducing something that I would call (please, forgive my lack of a better word) "prosthetic" it would be better to fix it and make it so that `stx-transfer` and other functions would throw a proper error when you try to send funds to a wrong address?
In the end, what's the point of adding such verification to a new functions if the most important ones don't have it?